airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -741,8 +749,6 @@ def trigger_reentry(self, context: Context, event: dict[str, Any]) -> Any:
     def _clean(self, event: dict[str, Any]):
         if event["status"] == "running":
             return
-        if self.get_logs:
```

Comment:
I have moved up this check because if any exception occurs here, the subsequent cleanup lines won't be executed.
This is consistent with what we had earlier.